tests/ui/EditTask.test.ts

```diff
@@ -520,3 +521,21 @@ describe('Exhaustive editing', () => {
         );
     });
 });
+
+describe('Edit Modal HTML snapshot tests', () => {
+    it('should match snapshot', () => {
+        const task = taskFromLine({ line: '- [ ] absolutely to do', path: '' });
+        const onSubmit = () => {};
+        const allTasks = [task];
+        const { container } = renderAndCheckModal(task, onSubmit, allTasks);
+
+        const modalHTML = container.innerHTML;
+        const prettyHTML = prettier.format(modalHTML, {
+            parser: 'html',
+            bracketSameLine: true,
+            htmlWhitespaceSensitivity: 'ignore',
+            printWidth: 120,
+        });
+        expect(prettyHTML).toMatchSnapshot();
```

Comment:
Thanks very much for doing this - it's going to be really useful.

All the but the expect line looks very similar to some code in `tests/Renderer/TaskLineRenderer.test.ts`.

Either in this PR, or soon after, it would be a good candidate for creating a a helper function, both for readability and consistency.

Personally, when comparing Jest-non-inline-snapshots and Approval Tests, I prefer Approval Tests:

- Jest does not delete snapshot files when their test is deleted - so snapshot files accumulate clutter
- When approval tests differ, a diff tool is opened, so for example some of the changes can be approved, and others not... and the diff tool will typically support syntax highlighting
- When Jest snapshot files differ, there is no diff tool, not all users will have WebStorm's easy ability to regenerate snapshots - and those that do have WebStorm may not be aware of the facility.

If this test were to use the same approval test call as in `tests/Renderer/TaskLineRenderer.test.ts` then this whole block of selected could could be replaced with a suitably named text helper function.

Reply:
> Jest does not delete snapshot files when their test is deleted - so snapshot files accumulate clutter

The same is of course true of `*.approved.*` files, but because they are right next to the source file in the IDE, it's much more likely that developers will remember to delete approved files...

The same applies to renaming test files, and renaming tests...